src/preprocess_info.cc

```diff
@@ -158,7 +195,162 @@ Napi::Value PreProcessInfo::GetMeanVariant(const Napi::CallbackInfo& info) {
+      new ie::LockedMemory<const void>(memory_meandata->rmap())));
+  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
+      env,
+      locked_memory_->as<ie::PrecisionTrait<ie::Precision::I8>::value_type*>(),
+      meandata->byteSize());
+
+  preprocess_channel.Set("meanData", buffer);
+  Napi::Value channelInfo = Napi::Value::From(env, preprocess_channel);
+
+  return channelInfo;
 }
+
+void PreProcessInfo::SetPreProcessChannel(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  if (info.Length() != 2) {
+    Napi::TypeError::New(env, "Wrong number of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[0].IsNumber()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  if (!info[1].IsObject()) {
+    Napi::TypeError::New(env, "Wrong type of arguments")
+        .ThrowAsJavaScriptException();
+    return;
+  }
+
+  size_t index = info[0].ToNumber().Int32Value();
+  Napi::Object newChannel = info[1].ToObject();
+
+  ie::PreProcessInfo& preInfo = _input_info->getPreProcess();
+
+  size_t numberOfChannels = preInfo.getNumberOfChannels();
+
+  if (numberOfChannels == 0) {
+    Napi::Error::New(env, "accessing pre-process when nothing was set.")
+        .ThrowAsJavaScriptException();
+    return;
+  };
+
+  if (index >= numberOfChannels) {
+    std::string errorLog =
+        "pre process index " + std::to_string(index) + " is out of bounds.";
+    Napi::Error::New(env, errorLog).ThrowAsJavaScriptException();
+    return;
+  };
+
+  ie::PreProcessChannel::Ptr channel = preInfo[index];
+  if (newChannel.Has("stdScale")) {
+    if (newChannel.Get("stdScale").IsNumber()) {
+      channel->stdScale = newChannel.Get("stdScale").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong stdScale").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanValue")) {
+    if (newChannel.Get("meanValue").IsNumber()) {
+      channel->meanValue = newChannel.Get("meanValue").ToNumber().FloatValue();
+    } else {
+      Napi::TypeError::New(env, "Wrong meanValue").ThrowAsJavaScriptException();
+    }
+  }
+
+  if (newChannel.Has("meanData")) {
+    if (!newChannel.Get("meanData").IsArrayBuffer()) {
+      Napi::TypeError::New(env, "Wrong meanData").ThrowAsJavaScriptException();
+    } else {
+      Napi::ArrayBuffer meanDataArray =
+          Napi::ArrayBuffer::ArrayBuffer(env, newChannel.Get("meanData"));
+      size_t bytelength = meanDataArray.ByteLength();
+      void* buffer = meanDataArray.Data();
+
+      try {
+        ie::SizeVector dims_vector = {bytelength, 1};
+        ie::TensorDesc tensor(ie::Precision::FP32, dims_vector, ie::Layout::HW);
+        ie::Blob::Ptr meanDataBlob = ie::make_shared_blob<float>(
+            tensor, reinterpret_cast<float_t*>(buffer), bytelength);
+        meanDataBlob->allocate();
+        channel->meanData = meanDataBlob;
```

Comment:
We need js code to specify the layout and size vector to this blob. So we may need to change the API as https://github.com/intel/inference-engine-node/issues/26#issuecomment-662778967 to accept a `MeanData` which contains both `TensorDesc` and `ArrayBuffer`. And please create the `Blob` and allocate its memory and copy the data from `ArrayBuffer` to this `Blob`.

Reply:
Done.